src/core/settings/qgssettings.h

```diff
@@ -79,36 +80,51 @@ class CORE_EXPORT QgsSettings : public QObject
-        static const inline char *ELEVATION_PROFILE = "elevation-profile";
-        static const inline char *CORE_LAYERTREE = "core/layer-tree";
-        static const inline char *STYLE_MANAGER = "app/style-manager";
-        static const inline char *FONTS = "fonts";
-        static const inline char *WMS = "wms";
-    };
+    static QgsSettingsTreeElement *treeRoot();
+
+    // only create first level here
+    static inline QgsSettingsTreeElement *sTreeApp = treeRoot()->createChildElement( QStringLiteral( "app" ) );
+    static inline QgsSettingsTreeElement *sTreeConnections = treeRoot()->createChildElement( QStringLiteral( "connections" ) );
+    static inline QgsSettingsTreeElement *sTreeCore = treeRoot()->createChildElement( QStringLiteral( "core" ) );
+    static inline QgsSettingsTreeElement *sTreeElevationProfile = treeRoot()->createChildElement( QStringLiteral( "elevation-profile" ) );
+    static inline QgsSettingsTreeElement *sTreeFonts = treeRoot()->createChildElement( QStringLiteral( "fonts" ) );
+    static inline QgsSettingsTreeElement *sTreeGeometryValidation = treeRoot()->createChildElement( QStringLiteral( "geometry_validation" ) );
+    static inline QgsSettingsTreeElement *sTreeGps = treeRoot()->createChildElement( QStringLiteral( "gps" ) );
+    static inline QgsSettingsTreeElement *sTreeGui = treeRoot()->createChildElement( QStringLiteral( "gui" ) );
+    static inline QgsSettingsTreeElement *sTreeLayerTree = treeRoot()->createChildElement( QStringLiteral( "layer-tree" ) );
+    static inline QgsSettingsTreeElement *sTreeLayout = treeRoot()->createChildElement( QStringLiteral( "layout" ) );
+    static inline QgsSettingsTreeElement *sTreeLocale = treeRoot()->createChildElement( QStringLiteral( "locale" ) );
+    static inline QgsSettingsTreeElement *sTreeMap = treeRoot()->createChildElement( QStringLiteral( "map" ) );
+    static inline QgsSettingsTreeElement *sTreeNetwork = treeRoot()->createChildElement( QStringLiteral( "network" ) );
+    static inline QgsSettingsTreeElement *sTreeQgis = treeRoot()->createChildElement( QStringLiteral( "qgis" ) );
+    static inline QgsSettingsTreeElement *sTreePlugins = treeRoot()->createChildElement( QStringLiteral( "plugins" ) );
+    static inline QgsSettingsTreeElement *sTreeProcessing = treeRoot()->createChildElement( QStringLiteral( "processing" ) );
+    static inline QgsSettingsTreeElement *sTreeSvg = treeRoot()->createChildElement( QStringLiteral( "svg" ) );
+    static inline QgsSettingsTreeElement *sTreeWms = treeRoot()->createChildElement( QStringLiteral( "wms" ) );
+
+    // sub levels
+    static inline QgsSettingsTreeElement *sTreeDigitizing = sTreeQgis->createChildElement( QStringLiteral( "digitizing" ) );
+
+#endif
+
+    /**
+     * Creates a settings tree element for the given \a pluginName
+     * \since QGIS 3.30
+     */
+    static QgsSettingsTreeElement *createPluginTreeElement( const QString &pluginName ) SIP_THROW( QgsSettingsException );
```

Comment:
Why are we using "element" in these object names instead of say "entry" which is used elsewhere? Element to me is associated with QDom* and XML .

("Node" could be another option which matches eg the layer tree classes)

Reply:
Here the elements are the basically building the tree structure, they are not the same things than the settings entries.

I have renamed to nodes, thanks for the suggestion.